presto-hive-common/src/main/java/com/facebook/presto/hive/MetadataUtils.java

```diff
@@ -155,4 +161,27 @@ private static Domain buildColumnDomain(ColumnHandle column, List<HivePartition>
 
         return Domain.onlyNull(type);
     }
+
+    /**
+     * Constructs the schema name, including catalog name if applicable.
+     *
+     * @param schemaName the original schema name
+     * @return the formatted schema name (Example - @catalog_name#schema_name)
+     */
+    public static String constructSchemaName(Optional<String> catalogName, @Nullable String schemaName)
+    {
+        if (!catalogName.isPresent() || DEFAULT_DATABASE.equals(schemaName) ||
+                (schemaName != null && schemaName.contains(CATALOG_DB_SEPARATOR))) {
+            return schemaName;
+        }
+
+        StringBuilder catalogDatabaseName = new StringBuilder()
+                .append(CATALOG_DB_THRIFT_NAME_MARKER)
+                .append(catalogName.get()) // Safe since we checked isPresent()
+                .append(CATALOG_DB_SEPARATOR);
+
+        catalogDatabaseName.append(schemaName == null ? "" : schemaName.isEmpty() ? DB_EMPTY_MARKER : schemaName);
```

Comment:
nit: Can we follow the above pattern instead?
```suggestion
                .append(CATALOG_DB_SEPARATOR)
                .append(schemaName == null ? "" : schemaName.isEmpty() ? DB_EMPTY_MARKER : schemaName);
```